### ISSUE TITLE
A11y_Windows Template Studio_Select a Project type_ColorContrast: Color contrast ratio of Keyboard focus outline is less than 3:1.

### ISSUE THREAD

**Issue by harshgangrade111** [microsoft/WindowsTemplateStudio#3901]
[“Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags:** #A11yWCAG2.1;#A11yTCS;#A11ySev2;#GH_WindowsTemplateStudio_Oct2020; #Windows Template Studio;#Win10;#FTP;#DesktopApp;#WCAG1.4.11;#AI4D;#NonBenchmark;#keyboard;#ColorContrast;#closed;

### **Environment Details:**
Application Name: Windows Template Studio (WinTS)
Templates version: 3.8.20273.1
Wizard version: 3.8.20273.1Date: 2020-06-17T21:13:20.174Z
Microsoft Visual Studio Enterprise 2019 Preview
Version 16.7.0 Preview 6.0
Windows Version: Windows10

### **Repro Steps:**
1.	Open Visual Studio 2019 
2.	Click on File from menu and then select new and then project.
3.	Select the "Windows template studio (WPF.Net Core)”. Click on next
4.	"Configure your new project" screen will appear. Click on create button.
5.	New WPF App screen will open.
6.	Navigate to “Project Type” List item present in  navigation pane with the help of Tab key and hit enter..
7.	Tab to “Blank” list Item.
8.	Start AI4D tool and check color contrast ratio of focus outline.

### **Actual:**
Color contrast ratio of Keyboard focus outline is less 3:1.

**Same issue is also observed with Following Scenarios:** 
•	Select a Design pattern
•	Add Pages
•	Add Features
•	Add Services
•	Select the Template You Want To Add
•	Template Details

### **Expected:**
Color contrast ratio of Keyboard focus outline should be more than or equal to 3:1 sop then it will be properly visible.

### **User Impact:**
Low vision users will not be able to see keyboard focus properly if color contrast if low..

**Comment by vmadurga** [comment]
The current focus outline color is **#9A9A9A** with contrast ratio 2.813:1.

We have to increase the current color contrast or using a new one.

If we check the color contrast using Accessibility Insights, the picked color may be different depending on the screen.

- Screen 1 (correct):

![imagen](https://user-images.githubusercontent.com/25466105/96857043-d2bcc500-145e-11eb-97c8-da842cb210ab.png)

- Screen 2 (incorrect):

![imagen](https://user-images.githubusercontent.com/25466105/96857088-e10ae100-145e-11eb-944b-65fdeb797e39.png)

**Comment by mrlacey** [comment]
My understanding is that the designs were approved by the Visual Studio design team.
- Was accessibility not considered as part of this review?
- Were incorrect colors used (in some places)?
- Does Visual Studio meet the contrast difference requirements?

> The current focus outline color is #9A9A9A with contrast ratio 2.813:1.

- Based on which theme?

- Why does the wizard code use hard coded color values instead of using the resources provided as part of the VSSDK? For example, I'd expect the active border to use [ActiveBorderKey](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.vsbrushes.activeborderkey?view=visualstudiosdk-2019#Microsoft_VisualStudio_Shell_VsBrushes_ActiveBorderKey)

**Comment by sibille** [comment]
That's correct, the designs were done and reviewed by the VS design team.

We use (according to the designs) VS colors and also define some custom colors in a Windows Template Studio pkgdef. One of these colors is the CardBorderHover color that is set to #9A9A9A as Victor mentioned (for Light theme).

Our suggestion is to tweak this color a bit to meet the Color Contrast and get approval from the design team for the new color.

**Comment by vmadurga** [comment]
We could use the following focus outline color: **#949494**

It meet the Color Contrast Ratio:
![imagen](https://user-images.githubusercontent.com/25466105/96899168-e08b3e00-1490-11eb-9dc4-cf75deca9655.png)

**Comment by vmadurga** [comment]
Screenshot using current border color (**#9A9A9A**):

![imagen](https://user-images.githubusercontent.com/25466105/97006468-379c1c00-1540-11eb-9b9f-123e4d864cf9.png)

Screenshot using proposal border color (**#949494**):

![imagen](https://user-images.githubusercontent.com/25466105/97009254-29e89580-1544-11eb-911b-ccfc4adb8784.png)

**Comment by harshgangrade111** [comment]
@sibille  @vmadurga  Issue is still repro in the latest dev-nightly version. Color contrast ratio of focus outline is less than 3;1.
![#3901](https://user-images.githubusercontent.com/60422426/98677660-4fb6cc80-2383-11eb-9370-29964a5c7276.png)

**Comment by sibille** [comment]
@harshgangrade111, this is not fixed yet, we're waiting for a design-review for our proposed solution. 
Once we fixed the issue, I'll add the done tag then and reassign to you to test and close.

**Comment by sibille** [comment]
Spoke to the design team and got the approval for using 949494 for our CardBorderHoverBrushKey

**Comment by sibille** [comment]
Verified in dev-nightly version
Wizard version: 0.22.20335.2

You can install the dev-nightly version from our nightly feed (https://www.myget.org/F/windows-template-studio-nightly/vsix/).
For instructions on how install WTS from the dev-nightly feed in Visual Studio see https://github.com/microsoft/WindowsTemplateStudio/blob/dev/docs/getting-started-extension.md#nightly--pre-release-instructions

**Comment by harshgangrade111** [comment]
@sibille as discuss on call. Issue is still repro in the provided latest Wizard version: 0.22.20337.1.
Color contrast ratio of keyboard focus outline is less than 3:1.

**Comment by harshgangrade111** [comment]
@sibille Issue does not repro now. Hence closing this bug.
